packages/spanner/src/transaction.js

```diff
@@ -223,25 +228,26 @@ Transaction.prototype.begin = function(callback) {
  * });
  *
  * //-
- * // If the callback is omitted, we'll return a Promise.
+ * // To use promises, simply return a Promise.
  * //-
- * database.runTransaction()
- *   .then(function(data) {
- *     var transaction = data[0];
+ * database.runTransaction(function(err, transaction) {
+ *   if (err) {
+ *     // Error handling omitted.
```

Comment:
What type of err would this be? Would the user's `.catch` block get executed with it, too, which I think is where they'd expect to handle it?

As a promise user, if you were using this method, would you think it would be easier to handle the success/failure of a commit *on the commit call itself*? I know we had some tricky requirements for this method with the retry rules, but given the chance to re-think it for a second, what about:

```js
database.runTransaction(function(err, transaction) {
  // handle err

  transaction.insert(...)

  transaction.commit()
    .then(handleSuccess, handleFailure)
})
```

This way, I don't think they'd have to return the promise, so the whole thing would feel more predictable. I believe `commit` (or any other transaction-terminating method) would still be able to retry the function passed to `runTransaction`. Because if it failed, it would not execute the callback, which would block the promise from resolving. Their promise pipeline would halt and we'd run the `runFn` again, and it would do this until it succeeds or hits a max-attempt limit.

WDYT? Is that possible?